scripts/astrokat-catalogue2obsfile.py

```diff
@@ -238,14 +260,21 @@ class BuildObservation(object):
     target_list: list
         A list of targets with the format
         'name=<name>, radec=<HH:MM:SS.f>,<DD:MM:SS.f>, tags=<tags>, duration=<sec>'
+        'name=<name>, gal=<deg float>,<deg float>, tags=<tags>, duration=<sec>'
+        'name=<name>, azel=<deg float>,<deg float>, tags=<tags>, duration=<sec>'
+        'name=<name>, special=<ephem_name>, tags=<tags>, duration=<sec>'
+        'name=<name>, xephem=<EDB string>, tags=<tags>, duration=<sec>'
 
     """
 
     def __init__(self, target_list):
         self.target_list = target_list
         self.configuration = None
 
-    def configure(self, instrument={}, obs_duration=None, lst=None):
+    def configure(self,
+                  instrument=None,
```

Comment:
Does changing this keyword from `instrument={}` to `None` have any special impact?

Reply:
Not sure about the effect on the rest of the method (that should be checked).  However, using `None` as a default kwarg value is a better approach in general.  Kwargs shouldn't be initialised with containers like lists and dicts, since the initialisation only happens why Python first imports the code, not on every call.

E.g.,
```python
>>> def bad_func(x={}):
...     print(x)
...     x['a'] = 1
... 
>>> bad_func()
{}
>>> bad_func()
{'a': 1}
>>> bad_func()
{'a': 1}
>>> bad_func({'b': 2})
{'b': 2}
>>> bad_func()
{'a': 1}
```

Reply:
I notice that the check for `if len(instrument) > 0:` will fail if `instrument` is `None`. 
```python
>>> len(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
```
Please fix, @bngcebetsha